e2e/test/scenarios/onboarding/search/search.cy.spec.js

```diff
@@ -263,6 +290,57 @@ describe("scenarios > search", () => {
           });
         });
       });
+
+      describe("created_by filter", () => {
+        beforeEach(() => {
+          // create a bunch of questions from a normal user, then we can query the question
+          // created by that user as an admin
+          cy.signInAsNormalUser();
+          for (const question of TEST_QUESTIONS) {
+            cy.createQuestion(question);
+          }
+          cy.signOut();
+
+          cy.signInAsAdmin();
+        });
+
+        it(`should filter results by user`, () => {
+          cy.visit("/");
+
+          getSearchBar().clear().type("e{enter}");
+          cy.wait("@search");
+
+          cy.findByTestId("created_by-search-filter").click();
+
+          popover().within(() => {
+            cy.findByText("Robert Tableton").click();
+            cy.findByText("Apply filters").click();
+          });
+          cy.url().should("contain", "created_by");
+
+          cy.findAllByTestId("search-result-item").each(result => {
+            cy.wrap(result).should("contain.text", "Robert's Question");
+          });
+        });
+
+        it("should remove type filter when `X` is clicked on filter", () => {
+          cy.visit(`/search?q=e&created_by=${NORMAL_PERSONAL_USER_ID}`);
+
+          cy.findByTestId("created_by-search-filter").within(() => {
+            cy.findByText("Robert Tableton").should("exist");
+            cy.findByLabelText("close icon").click();
+          });
+
+          // Check that we're getting elements from other users by checking for other types, since all assets
+          // for the user we're querying are questions
+          cy.findAllByTestId("result-link-info-text").then($elements => {
+            const textContent = new Set(
+              $elements.toArray().map(el => el.textContent),
+            );
+            expect(textContent.size).to.be.greaterThan(1);
```

Comment:
1. It's not clear why do we need to use a `Set` here
2. Perhaps adding another assertion right after `cy.visit()` would make it clear how displayed values are expected to change when clicking X

Reply:
On (1): This will change in the future with some of the changes when we get user info in the search results. For now, I'm using the types as a proxy for "user-created items," and the `result-link-info-text` elements contain the type information. We're using a `Set` here to check if we're seeing search results that aren't just questions (so creating a list of unique elements).

So, for example, Robert creates 5 Question assets. If we filter by `created_by=[Robert's ID]` then we will only have 5 Questions, where each search result will have `Question in [collection]`. If we remove that filter, we expect to have the databases, dashboards, and other assets that the Admin user has created, so we expect to see `Dashboard in...`, `Database`, etc

Again this will change when we implement the next milestone, where search results will have the user's name available to query

Reply:
On (2): could you clarify what you mean? I'm interpreting it as `check that there's only 5 elements and then click X` but just want to make sure

Reply:
> On (2): could you clarify what you mean? I'm interpreting it as `check that there's only 5 elements and then click X` but just want to make sure

Initially I suggested to have this code right after `cy.visit` call:

```js
          cy.findAllByTestId("result-link-info-text").then($elements => {
            const textContent = new Set(
              $elements.toArray().map(el => el.textContent),
            );
            expect(textContent.size).to.equal(1);
          });
```

It was not obvious to me how removing the filter will affect `textContent.size`.

But after running the test myself I'd like to update my suggestion:
- rename `$elements` to `$resultsDescriptions` or `$descriptions`
- rename `textContent` to `differentDescriptions` or `differentResultsDescriptions`

It should help understanding the code without having to run the test.

Non blocking though.

Reply:
Gotcha, I went with: `$resultTypeDescriptions` and `uniqueTypeDescriptions` for a little extra flavor